examples/basic-typescript/pages/_app.tsx

```diff
@@ -0,0 +1,26 @@
+import { AppProps } from 'next/app'
+import { Inter } from 'next/font/google'
+import { useEffect } from 'react'
+
+import clsx from 'clsx'
+
+import '../styles/globals.css'
+
+const inter = Inter({
+  subsets: ['latin'],
+  variable: '--font-inter',
+})
+
+function MyApp({ Component, pageProps }: AppProps) {
+  useEffect(() => {
+    document.body.classList.add(inter.variable)
+  }, [])
```

Comment:
Why do we need to add the class to the body here? Won't this cause FOUC?

Reply:
This is just for modals and anything that is portaled out of the `<main>` element.

Reply:
Got it. Can this not be added to the `body` in `_document`, then? It's how I see `create-next-app` do it for the App Router, it puts it in the root layout's `body` element.

Reply:
Or maybe `next/font` can't be used in `_document`... They don't seem to do it with `create-next-app` when using the pages router